lib/logstash/util/wrapped_synchronous_queue.rb

```diff
@@ -0,0 +1,25 @@
+module LogStash; module Util
+  class WrappedSynchronousQueue
+    java_import java.util.concurrent.SynchronousQueue
+    java_import java.util.concurrent.TimeUnit
+
+    def initialize()
+      @queue = java.util.concurrent.SynchronousQueue.new()
+    end
+
+    def push(obj)
+      @queue.put(obj)
+    end
+    alias_method(:<<, :push)
```

Comment:
Thinking about it, I'd rather keep the surface area small for this. If you feel strongly about it I can add it, but that's my preference.